style/javascript.md

```diff
@@ -552,3 +553,53 @@ Yes:
+
+Method | Use...                                | ...instead of
+--------- | ------------------------------------- | ----------------------
+bind | `fn.bind(someObj, args)` | `_.bind(fn, someObj, args)`
+bind | `() => { ... }` | `_.bind(function() { ... }, this)`
+bindAll | `obj.method = obj.method.bind(someObj);` <sup>[1](#u1)</sup> | `_.bindAll(someObj, "method")`
+clone | No alternative at the moment! |
+debounce | Our custom lodash build. <sup>[2](#u2)</sup> |
+defer | `setTimeout(fn, 0);` | `_.defer(fn);`
+delay | `setTimeout(fn, 2000);` | `_.delay(fn, 2000);`
+each | `array.forEach(fn)` | `_.each(array, fn)`
+each | `for (const val of array) {}` | `_.each(array, fn)`
+each | `for (const [key, val] of Object.entries(obj)) {}` | `_.each(obj, fn)`
+extend | `{...options, prop: 1}` | `_.extend({}, options, {prop: 1})`
+extend | `{...defaultOptions, ...options}` | `_.extend(defaultOptions, options || {})`
+extend | `Object.assign(json, this.model.toJSON())` | `_.extend(json, this.model.toJSON())`
+filter | `array.filter(checkFn)` | `_.filter(array, checkFn)`
+has | `array.includes(value)` | `_.has(array, value)`
+has | `obj.hasOwnProperty(value)` <sup>[3](#u3)</sup> | `_.has(obj, value)`
+isArray | `Array.isArray(someObj)` | `_.isArray(someObj)`
+isFunction | `typeof fn === "function"` | `_.isFunction(fn)`
+isString | `typeof obj === "string"` | `_.isString(obj)`
+keys | `Object.keys(obj)` | `_.keys(obj)`
+last | `someArray[someArray.length - 1]` | `_.last(someArray)`
+last | `someArray.pop()` <sup>[4](#u4)</sup> | `_.last(someArray)`
+map | `array.map(mapFn)` | `_.map(array, mapFn)`
+max | `Math.max(...array)` | `_.max(array)`
+omit | `array.filter(prop => !props.includes(prop))` | `_.omit(array, props)`
+omit | <pre>Object.keys(obj).reduce((result, prop) => {<br>&nbsp;&nbsp;&nbsp;&nbsp;if (!props.includes(prop)) {<br>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;result[prop] = attrs[prop];<br>&nbsp;&nbsp;&nbsp;&nbsp;}<br>}, {})</pre> | `_.omit(obj, props)`
+once | `$(...).one("click", ...)` | `$(...).on("click", _.once(...))`
+once | <pre>{<br>&nbsp;&nbsp;&nbsp;&nbsp;method: () => {<br>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;if (this._initDone) { return; }<br>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;this._initDone = true;<br>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;...<br>&nbsp;&nbsp;&nbsp;&nbsp;}<br>}</pre>| `{ method: _.once(() => { ... }) }`</pre>
+once | <pre>var getResult = () => {<br>&nbsp;&nbsp;&nbsp;&nbsp;let val = $.when(...).then(...);<br>&nbsp;&nbsp;&nbsp;&nbsp;getResult = () => val;<br>&nbsp;&nbsp;&nbsp;&nbsp;return val;<br>};</pre> | <pre>var getResult = _.once(() => {<br>&nbsp;&nbsp;&nbsp;&nbsp;return $.when(...).then(...);<br>});</pre>
+object | <pre>Object.entries(obj).reduce((result, [key, val]) => {<br>&nbsp;&nbsp;&nbsp;&nbsp;result[key] = value;<br>&nbsp;&nbsp;&nbsp;&nbsp;return result;<br>})</pre> | <pre>_.object(_.map(obj, (val, key) => {<br>&nbsp;&nbsp;&nbsp;&nbsp;return [key, value];<br>})</pre>
```

Comment:
It's probably possible to use backticks instead of literal <pre> tags, and GitHub should treat the `stuff within backticks` as code/pre and not parse the underscores inside as markdown.

On my phone and not testing this at the moment, but there should be a way to get this to work.



> On Oct 17, 2015, at 4:05 PM, Craig Silverstein <notifications@github.com> wrote:
> 
> In style/javascript.md:
> 
> > +has | `array.includes(value)` | `_.has(array, value)`
> > +has | `obj.hasOwnProperty(value)` <sup>[3](#u3)</sup> | `_.has(obj, value)`
> > +isArray | `Array.isArray(someObj)` | `_.isArray(someObj)`
> > +isFunction | `typeof fn === "function"` | `_.isFunction(fn)`
> > +isString | `typeof obj === "string"` | `_.isString(obj)`
> > +keys | `Object.keys(obj)` | `_.keys(obj)`
> > +last | `someArray[someArray.length - 1]` | `_.last(someArray)`
> > +last | `someArray.pop()` <sup>[4](#u4)</sup> | `_.last(someArray)`
> > +map | `array.map(mapFn)` | `_.map(array, mapFn)`
> > +max | `Math.max(...array)` | `_.max(array)`
> > +omit | `array.filter(prop => !props.includes(prop))` | `_.omit(array, props)`
> > +omit | <pre>Object.keys(obj).reduce((result, prop) => {<br>&nbsp;&nbsp;&nbsp;&nbsp;if (!props.includes(prop)) {<br>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;result[prop] = attrs[prop];<br>&nbsp;&nbsp;&nbsp;&nbsp;}<br>}, {})</pre> | `_.omit(obj, props)`
> > +once | `$(...).one("click", ...)` | `$(...).on("click", _.once(...))`
> > +once | <pre>{<br>&nbsp;&nbsp;&nbsp;&nbsp;method: () => {<br>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;if (this._initDone) { return; }<br>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;this._initDone = true;<br>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;...<br>&nbsp;&nbsp;&nbsp;&nbsp;}<br>}</pre>| `{ method: _.once(() => { ... }) }`</pre>
> > +once | <pre>var getResult = () => {<br>&nbsp;&nbsp;&nbsp;&nbsp;let val = $.when(...).then(...);<br>&nbsp;&nbsp;&nbsp;&nbsp;getResult = () => val;<br>&nbsp;&nbsp;&nbsp;&nbsp;return val;<br>};</pre> | <pre>var getResult = _.once(() => {<br>&nbsp;&nbsp;&nbsp;&nbsp;return $.when(...).then(...);<br>});</pre>
> > +object | <pre>Object.entries(obj).reduce((result, [key, val]) => {<br>&nbsp;&nbsp;&nbsp;&nbsp;result[key] = value;<br>&nbsp;&nbsp;&nbsp;&nbsp;return result;<br>})</pre> | <pre>_.object(_.map(obj, (val, key) => {<br>&nbsp;&nbsp;&nbsp;&nbsp;return [key, value];<br>})</pre>
> This one doesn't render right, sadly: the _'s are taken as markup directives, not literals.
> 
> —
> Reply to this email directly or view it on GitHub.
>